systemtest/src/test/java/io/strimzi/systemtest/security/custom/CustomCaST.java

```diff
@@ -0,0 +1,591 @@
+     */
+    private void prepareTestCaWithBundleAndKafkaCluster(final ExtensionContext extensionContext, final SystemTestCertHolder certificateAuthority, final TestStorage ts) {
+        // 1. Prepare correspondent Secrets from generated custom CA certificates
+        //  a) Cluster or Clients CA
+        certificateAuthority.prepareCustomSecretsFromBundles(ts.getNamespaceName(), ts.getClusterName());
+
+        //  b) if Cluster CA is under test - we generate custom Clients CA (it's because in our Kafka configuration we
+        //     specify to not generate CA automatically. Thus we need it generate on our own to avoid issue
+        //     (f.e., Clients CA should not be generated, but the secrets were not found.)
+        if (certificateAuthority.getCaCertSecretName().equals(KafkaResources.clusterCaCertificateSecretName(ts.getClusterName())) &&
+            certificateAuthority.getCaKeySecretName().equals(KafkaResources.clusterCaKeySecretName(ts.getClusterName()))) {
+            final SystemTestCertHolder clientsCa = new SystemTestCertHolder(extensionContext,
+                "CN=" + extensionContext.getRequiredTestClass().getSimpleName() + "ClientsCA",
+                KafkaResources.clientsCaCertificateSecretName(ts.getClusterName()),
+                KafkaResources.clientsCaKeySecretName(ts.getClusterName()));
+            clientsCa.prepareCustomSecretsFromBundles(ts.getNamespaceName(), ts.getClusterName());
+        } else {
+            // otherwise we generate Cluster CA
+            final SystemTestCertHolder clusterCa = new SystemTestCertHolder(extensionContext,
+                "CN=" + extensionContext.getRequiredTestClass().getSimpleName() + "ClusterCA",
+                KafkaResources.clusterCaCertificateSecretName(ts.getClusterName()),
+                KafkaResources.clusterCaKeySecretName(ts.getClusterName()));
+            clusterCa.prepareCustomSecretsFromBundles(ts.getNamespaceName(), ts.getClusterName());
+        }
+
+        // 2. Create a Kafka cluster without implicit generation of CA
+        resourceManager.createResource(extensionContext, KafkaTemplates.kafkaEphemeral(ts.getClusterName(), 1)
+            .editOrNewSpec()
+                .withNewClientsCa()
+                    .withRenewalDays(5)
+                    .withValidityDays(20)
+                    .withGenerateCertificateAuthority(false)
+                .endClientsCa()
+            // TODO: If I un-comment this and run testReplacingCustomClusterKeyPairToInvokeRenewalProcess test it will fail
+            //  because only ZooKeeper pod will trigger RollingUpdate, Kafka pod has some errors:
+            //  -> https://gist.github.com/see-quick/d32c569572ae396597ce88394dc46fed
+//
+//               .withNewClusterCa()
+//                    .withRenewalDays(5)
+//                    .withValidityDays(20)
+//                    .withGenerateCertificateAuthority(false)
+//                .endClusterCa()
```

Comment:
Currently, as the code is written:
**Clients Ca**
 I verify that replacing the custom `Clients Ca` public and private key with different custom `Clients Ca` public and private key will trigger RollingUpdate of Kafka cluster.
**Cluster Ca**
When I un-comment these lines and let CO not generate CA for me (i have to do it myself). I create it and also replace it with a different `Cluster Ca` which does not trigger `RollingUpdate` of all components but only the `ZooKeeper` pod.

Reply:
@ppatierno ^^

Reply:
Well the second one is really strange because my manual tests as well as @im-konge ones seem to show that rolling happens. If not happening I haven't never released the fix because this is actually the problem it fixes.
It would be interesting to get logs in the CO when after the ZK roll, it doesn't rool Kafka.

Reply:
@ppatierno

Reply:
Are you using a different CN for the new certificate? This is what Strimzi does automatically by adding suffixes like `v1` to the first CA cert, then `v2` on renew and so on.

Reply:
No, I am using the same CN (i.e., for Cluster`"CN=" + extensionContext.getRequiredTestClass().getSimpleName() + "ClusterCA"` or for Clients`"CN=" + extensionContext.getRequiredTestClass().getSimpleName() + "ClientsCA"`). But I am generating a new root CA, intermediate CA. Could this be a problem?

Reply:
edit:
Okay, I have updated the CN for a new certificate with the suffix `v2` and now it's working. But for only `Cluster`, for `Clients CA` its passing without adding the additional `v2`.

Reply:
Maybe that's a different case but I would behave the same way as when Strimzi is handling the renewal. It also adds a `vX` suffix to the CN for both Clients and Cluster CA.

Reply:
Updated )